test/kafka-ingest-open-loop/mzcompose.py

```diff
@@ -0,0 +1,187 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import time
+
+import pg8000
+
+from materialize.mzcompose import Composition, WorkflowArgumentParser, say
+from materialize.mzcompose.services import (
+    Kafka,
+    Kgen,
+    Materialized,
+    Testdrive,
+    Zookeeper,
+)
+
+mz_persist = Materialized(
+    name="materialized_persist",
+    hostname="materialized",
+    workers=4,
+    timestamp_frequency="1s",
+    options="--persistent-user-tables --persistent-kafka-upsert-source --disable-persistent-system-tables-test",
+)
+
+mz_default = Materialized(workers=4, timestamp_frequency="1s")
+
+prerequisites = ["zookeeper", "kafka"]
+
+SERVICES = [
+    Zookeeper(),
+    Kafka(),
+    mz_default,
+    mz_persist,
+    # This instance of Testdrive is used to initialize the benchmark, and needs to have a
+    # known seed.
+    Testdrive(no_reset=True, seed=1),
+    Kgen(),
+]
+
+
+def query_materialize(
+    c: Composition,
+    mz_service: str,
+) -> int:
+    host_port = c.default_port(mz_service)
+    query = "SELECT * FROM num_records_ingested"
+
+    try:
+        conn = pg8000.connect(
+            host="localhost", port=host_port, user="materialize", timeout=1
+        )
+        conn.autocommit = True
+
+        cursor = conn.cursor()
+        cursor.execute("SELECT * FROM load_test_materialization_frontier")
+        row = cursor.fetchone()
+        if row is None or len(row) != 1 or row[0] is None:
+            return 0
+        timestamp = int(row[0])
+        if timestamp == 0:
+            return 0
+        cursor.execute(f"SELECT * FROM records_ingested AS OF {timestamp}")
+        row = cursor.fetchone()
+        if row is None or len(row) != 1 or row[0] is None:
+            return 0
+        return int(row[0])
+    except Exception as e:
```

Comment:
`except Exception` is bad practice because it'll catch all sorts of things besides database errors. Can you figure out what more specific exception pg8000 is raising and catch that instead?

Reply:
done. it was an `InterfaceError`.

I had to import it like this:

```
from pg8000 import InterfaceError # type: ignore
```

to make pycheck happy.